webboot/webboot.go

```diff
@@ -26,32 +33,102 @@ var (
 	ipv4      = flag.Bool("ipv4", true, "use IPV4")
 	ipv6      = flag.Bool("ipv6", true, "use IPV6")
 	osSystems = map[string]webboot.Distro{
-		"TinyCore": webboot.Distro{"/boot/vmlinuz64 ", "/boot/initrd", "--reuse-cmdline", "https:louis.com"},
-		"Ubuntu":   webboot.Distro{"/boot/vmlinuz ", "/boot/init", "--reuse-cmdline", "https:louis.com"},
+		"tinycore": webboot.Distro{"/boot/vmlinuz64 ", "/boot/initrd", "--reuse-cmdline", "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso"},
+		"ubuntu":   webboot.Distro{"/boot/vmlinuz ", "/boot/init", "--reuse-cmdline", "https://releases.ubuntu.com/bionic/ubuntu-18.04.2-desktop-amd64.iso"},
 	}
 )
 
+//wgetArgs takes in the arg user passes e.g. name of ISO or URL to an ISO
+//returns an io.ReadCloser, URL link, filename, and err
+//io.ReadCloser and err is returned from wget
+// if the arg is a name of an URL found in the osSystems, the Download Link specificed to the name of ISO is used for wget & urlToName to generate a name
+// and the Download Link then is returned instead of the passed-in URL
+func wgetArgs(arg string) (io.ReadCloser, string, string, error) {
+	var URL, filename string
+	//checks if arg is a key in osSystems map & a valid name to an ISO
+	if u, ok := osSystems[arg]; ok {
+		//if found in map, it finds the ISO's download link to use as URL for wget
+		URL = u.DownloadLink
+		filename = arg
+	} else {
+		//calls urlToName to change the filename to be the base of the link
+		filename = urlToName(arg)
+	}
+	read, err := wget(URL)
+	return read, URL, filename, err
+}
+
+func wget(URL string) (io.ReadCloser, error) {
+	//gets a response from the specified URL
+	resp, err := http.Get(URL)
+	if err != nil {
+		return nil, err
+	}
+	//checks if request to receive URL succeeded and the transmission level is successful on http level
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+//func writer copies from an io.ReadCloser into filename.
+func writer(read io.ReadCloser, filename string) (string, error) {
+	w, err := os.Create(filename)
+	if err != nil {
+		return "", err
+	}
+
+	defer w.Close()
+
+	_, err = io.Copy(w, read)
+	return filename, err
```

Comment:
Yes, I see what you are saying. I will only return an error from the writer function() since I do not use the filename in main after it was being returned.